test/suites/storage.sh

```diff
@@ -819,3 +831,60 @@ test_storage() {
   LXD_DIR="${LXD_DIR}"
   kill_lxd "${LXD_STORAGE_DIR}"
 }
+
+test_storage_volume_snapshots() {
+  ensure_import_testimage
+
+  # shellcheck disable=2039
+  local LXD_STORAGE_DIR lxd_backend
+
+  lxd_backend=$(storage_backend "$LXD_DIR")
+  LXD_STORAGE_DIR=$(mktemp -d -p "${TEST_DIR}" XXXXXXXXX)
+  chmod +x "${LXD_STORAGE_DIR}"
+  spawn_lxd "${LXD_STORAGE_DIR}" false
+
+  # edit storage and pool description
+  # shellcheck disable=2039
+  local storage_pool storage_volume
+  storage_pool="lxdtest-$(basename "${LXD_DIR}")-pool"
+  storage_volume="${storage_pool}-vol"
+
+  lxc storage create "$storage_pool" "$lxd_backend"
+  lxc storage volume create "${storage_pool}" "${storage_volume}"
+  lxc launch testimage c1 -s "${storage_pool}"
+  lxc storage volume attach "${storage_pool}" "${storage_volume}" c1 /mnt
+  # Create file on volume
+  echo foobar > "${TEST_DIR}/testfile"
+  lxc file push "${TEST_DIR}/testfile" c1/mnt/testfile
+
+  # Validate file
+  lxc exec c1 -- test -f /mnt/testfile
+  [ "$(lxc exec c1 -- cat /mnt/testfile)" = 'foobar' ]
+
+  lxc storage volume detach "${storage_pool}" "${storage_volume}" c1
+  # This will create a snapshot named 'snap0'
+  lxc storage volume snapshot "${storage_pool}" "${storage_volume}"
```

Comment:
Might be worth testing `lxc storage volume list` and `lxc storage volume info` on the snapshot at this point for good measure.